src/base/mapreduce.jl

```diff
@@ -33,22 +33,22 @@ function reduce_rep_def(op, z, lvl::HollowData, idx, idxs...)
     end
 end
 
-reduce_rep_def(op, z, lvl::SparseData, idx::Drop, idxs...) = ExtrudeData(reduce_rep_def(op, z, lvl.lvl, idxs...))
+reduce_rep_def(op, z, lvl::SparseData, idx::Drop{Idx}, idxs...) where {Idx} = ExtrudeData(reduce_rep_def(op, z, lvl.lvl, idxs...), Idx)
```

Comment:
Does Idx need to be a type parameter of Drop? I think it would be okay for it to be in the runtime here.